src/Okta.Sdk/OktaClient.cs

```diff
@@ -91,27 +124,10 @@ public OktaClient(OktaClientConfiguration apiClientConfiguration = null, ILogger
         /// <param name="serializer">The JSON serializer to use, if any. Using the <c>DefaultSerializer</c> is still strongly recommended since it has all the behavior this SDK needs to work properly.
         /// If a custom serializer is used, the developer is responsible to add the required logic for this SDK to continue working properly. See <see cref="DefaultSerializer"/> to check out what settings can be configured.
         /// </param>
-        public OktaClient(OktaClientConfiguration apiClientConfiguration, HttpClient httpClient, ILogger logger = null, IRetryStrategy retryStrategy = null, ISerializer serializer = null)
+        /// <param name="oAuthTokenProvider">The custom token provider to renew an access token when it expires. Only works with <see cref="AuthorizationMode.BearerToken"/>.</param>
+        public OktaClient(OktaClientConfiguration apiClientConfiguration, HttpClient httpClient, ILogger logger = null, IRetryStrategy retryStrategy = null, ISerializer serializer = null, IOAuthTokenProvider oAuthTokenProvider = null)
+        : this(apiClientConfiguration, oAuthTokenProvider, httpClient, logger, retryStrategy, serializer)
         {
-            Configuration = GetConfigurationOrDefault(apiClientConfiguration);
-            OktaClientConfigurationValidator.Validate(Configuration);
-
-            logger = logger ?? NullLogger.Instance;
-            serializer = serializer ?? new DefaultSerializer();
-
-            var resourceFactory = new ResourceFactory(this, logger);
-            IOAuthTokenProvider oAuthTokenProvider = (Configuration.AuthorizationMode == AuthorizationMode.PrivateKey) ? new DefaultOAuthTokenProvider(Configuration, resourceFactory, logger: logger) : NullOAuthTokenProvider.Instance;
-            var requestExecutor = new DefaultRequestExecutor(Configuration, httpClient, logger, retryStrategy, oAuthTokenProvider);
-
```

Comment:
Do we wanna remove this whole section?

Reply:
Yes, we had two constructors with almost identical code, having them can be potential problem. 
I moved everything that can be moved to another protected constructor, it's turned out entire constructors could be moved and they are now empty